tests/devices/qubit/test_simulate.py

```diff
@@ -1178,3 +1179,68 @@ def test_qinfo_tf(self):
 
         grad5 = grad_tape.jacobian(results[5], phi)
         assert qml.math.allclose(grad5, expected_grads[5])
+
+
+ml_frameworks_list = [
+    "numpy",
+    pytest.param("autograd", marks=pytest.mark.autograd),
+    pytest.param("jax", marks=pytest.mark.jax),
+    pytest.param("torch", marks=pytest.mark.torch),
+    pytest.param("tensorflow", marks=pytest.mark.tf),
+]
+
+
+# pylint:disable=too-few-public-methods
+class TestMidCircuitMeasurements:
+    """Unit tests for simulating mid-circuit measurements."""
+
+    @pytest.mark.parametrize("ml_framework", ml_frameworks_list)
+    @pytest.mark.parametrize(
+        "postselect_mode", [None, "hw-like", "pad-invalid-samples", "fill-shots"]
+    )
+    def test_simulate_one_shot_native_mcm(self, ml_framework, postselect_mode):
+        """Unit tests for simulate_one_shot_native_mcm"""
+
+        with qml.queuing.AnnotatedQueue() as q:
+            qml.RX(np.pi / 4, wires=0)
+            m = qml.measure(wires=0, postselect=0)
+            qml.RX(np.pi / 4, wires=0)
+
+        circuit = qml.tape.QuantumScript(q.queue, [qml.expval(qml.Z(0)), qml.sample(m)], shots=[1])
+
+        rng = np.random.default_rng(1)
+        rng_keys = [np.random.default_rng(x) for x in rng.integers(0, 2**32, size=100)]
+
+        n_shots = 100
+        mcm_results = []
+        terminal_results = []
+        for i in range(n_shots):
+            result = simulate_one_shot_native_mcm(
+                circuit,
+                n_shots,
+                rng=rng_keys[i],
+                interface=ml_framework,
+                postselect_mode=postselect_mode,
+            )
+            terminal_results.append(result[0])
+            mcm_results.append(result[1])
+
+        if postselect_mode == "fill-shots":
+            assert all(ms == 0 for ms in mcm_results)
+            ttest_result = ttest_1samp(terminal_results, np.cos(np.pi / 4))
+            assert ttest_result.pvalue > 0.05
```

Comment:
Is this a sufficient criterion for assertion here?
Should we consider whether the statistic value is also important, and whether the null rejection threshold is valid for our use-case.

Same for below uses too.

Reply:
It seems like maybe the people involved here should have a quick chat about what heuristics are sensible for us to use. Admittedly, my investigation during hack week was minimal considering the potential scope of its use, so it would probably be helpful for everyone involved to standardize how we decide on which test to use, what threshold to use, etc.